client/dashboard/me/monetize-subscriptions/index.tsx

```diff
@@ -0,0 +1,121 @@
+import { __ } from '@wordpress/i18n';
+import { useMemo, useState } from 'react';
+import { DataViewsCard } from '../../components/dataviews-card';
+import FlashMessage from '../../components/flash-message';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import { adjustDataViewFieldsForWidth } from '../../utils/dataviews-width';
+import { useMonetizeFieldDefinitions } from './dataviews';
+import { getMonetizeSubscriptionsPageTitle, getMonetizeSubscriptionUrl } from './urls';
+
+const defaultPerPage = 10;
+
+const monetizeDesktopFields = [ 'status' ];
+const monetizeMobileFields: string[] = [];
+
+const monetizeDataView: View = {
+	type: 'table',
+	page: 1,
+	perPage: defaultPerPage,
+	titleField: 'product',
+	showTitle: true,
+	mediaField: 'site',
+	showMedia: true,
+	descriptionField: 'description',
+	showDescription: true,
+	fields: monetizeDesktopFields,
+	sort: {
+		field: 'product',
+		direction: 'desc',
+	},
+	layout: {},
+};
+
+function MonetizeSubscriptions() {
+	const [ currentView, setView ] = useState( monetizeDataView );
+	const ref = useResizeObserver( ( entries ) => {
+		const firstEntry = entries[ 0 ];
+		if ( firstEntry ) {
+			adjustDataViewFieldsForWidth( {
+				width: firstEntry.contentRect.width,
+				setView,
+				wideFields: monetizeDesktopFields,
+				desktopFields: monetizeDesktopFields,
+				mobileFields: monetizeMobileFields,
+			} );
+		}
+	} );
+
+	const monetizeDataFields = useMonetizeFieldDefinitions();
+	const { data: monetizeSubscriptions, isLoading: isLoadingMonetize } = useQuery(
+		monetizeSubscriptionsQuery()
+	);
+	const navigate = useNavigate();
+
+	const actions = useMemo(
+		() => [
+			{
+				id: 'manage-purchase',
+				label: __( 'Manage purchase' ),
+				isEligible: ( item: MonetizeSubscription ) => Boolean( item.ID ),
+				callback: ( items: MonetizeSubscription[] ) => {
+					const subscriptionId = items[ 0 ].ID;
+					if ( ! subscriptionId ) {
+						// eslint-disable-next-line no-console
+						console.error( 'Cannot display manage purchase page for subscription without ID' );
+						return;
+					}
+					navigate( {
+						to: getMonetizeSubscriptionUrl( subscriptionId ),
+					} );
+				},
+			},
+		],
+		[ navigate ]
+	);
+
+	const { data: adjustedMonetizeSubscriptions, paginationInfo } = useMemo( () => {
+		return filterSortAndPaginate( monetizeSubscriptions ?? [], currentView, monetizeDataFields );
+	}, [ monetizeSubscriptions, currentView, monetizeDataFields ] );
+
+	const getItemId = ( item: MonetizeSubscription ) => {
+		return item.ID;
+	};
+
+	return (
+		<PageLayout
+			size="large"
+			header={ <PageHeader title={ getMonetizeSubscriptionsPageTitle() } /> }
+		>
+			<div ref={ ref }>
+				{ /* Show a flash message if the URL contains ?showSuccessRemoved=true when a subscription is removed */ }
+				<FlashMessage
+					param="showSuccessRemoved"
```

Comment:
By having a single `FlashMessage` component, I was hoping that we could start to use a consistent query param name throughout the hosting dashboard. Is there a reason we couldn't use `?updated=product-removed` query param?

Is there a better query name than `updated` that you think we could use instead?

Reply:
Oh. I see. I can change that as well. I also thought `updated` was too generic...
Also, I'll add an helper to FlashMessage to add the right parameter to the URL. This way there is no need to hardcode `updated` everywhere.

Reply:
See https://github.com/Automattic/wp-calypso/pull/105695#discussion_r2367905630